Yosemite/Yosemite/Model/Enums/CardPresentPaymentsOnboardingState.swift

```diff
@@ -77,6 +82,8 @@ extension CardPresentPaymentOnboardingState {
             return nil
         case .selectPlugin:
             return "multiple_plugins_installed"
+        case .deactivateStripe:
+            return "stripe_should_be_deactivated"
```

Comment:
I couldn't find a source for these tracking strings other than this PR https://github.com/woocommerce/woocommerce-android/pull/4562 . Is there a table that I can follow/update for this new string? @malinajirka

Reply:
[Here are all the onboarding errors with the corresponding events](https://github.com/woocommerce/woocommerce-android/blob/1833f7994d21851e26c6dc6164000f8fa1033e59/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/cardreader/CardReaderTracker.kt#L121) we track on Android.

It's also a good practice to update the google sheet table - I'll share a link in a dm.

---------------------

I don't want to block this PR and I might be missing some context. However, I'm not sure I understand why we force the user to disable the Stripe plugin in this scenario? Could we perhaps just tell the app to use wcpay endpoints and completely ignore the fact that Stripe plugin is installed? AFAIK there isn't any conflict between the plugins so asking the user to disable an unsupported plugin feels unnecessary. Wdyt?

Reply:
Thanks so much for the context 🙇 !

> we could ignore the fact that both are installed and continue with WCPay regardless, so avoid showing an onboarding failure. I don't think that would be a good idea with our current design, since we are likely to support either (but not both) one day soon

I believe this might not be a valid concern anymore. We plan to let the user choose which plugin they want to use when the app detects both plugins are installed: More context on pdfdoF-FO-p2#comment-1537.


I think we might want to consider the following:
1. Use WCPay in Canada no matter if Stripe is installed or not.
2. Implement support for letting the user choose which plugin they want to use (show it just in US until we support Stripe in Canada).


P.S. If we decide to merge this PR, I'd at least consider removing `stripe_` from the event. Including the plugin name in the events bit us in the past (eg. we need to have to separate events for `stripe_not_setup` and `wcpay_not_setup` instead of having a single event `plugin_not_setup` with a plugin_type property). Wdyt?

Reply:
Thinking about this more, perhaps we could
1. Merge this PR.
2. Mirror it on Android.
3. Implement support for letting the user choose + remove the screen introduced in this PR.

We'd avoid the discussion paralysis and implementing it on Android should be pretty straightforward.

Wdyt?

Reply:
Thanks @malinajirka for your feedback!

> P.S. If we decide to merge this PR, I'd consider removing stripe_ from the event and using generic plugin_should_be_deactivated.
> Including the plugin name in the events bit us in the past (eg. we need to have two separate events for stripe_not_setup and wcpay_not_setup instead of having a single event plugin_not_setup with a plugin_type property). Wdyt?

My first approach was exactly that, to create a generic event that could handle any kind of plugin, something like:
`pluginShouldBeDeactivated(plugin: CardPresentPaymentsPlugin)`
Unfortunately making it so generic made the implementation of the view a bit too complex:
The view currently has the title "In-Person Payments are processed through WooCommerce Payments." If we make the case generic we would have to make the view title generic as well and pass the plugin that should remain as parameter, otherwise we could have the strange case where we pass the WCPay as parameter `pluginShouldBeDeactivated(plugin: .wcPay)`, ending up in something like:
Title: "In-Person Payments are processed through WooCommerce Payments."
Message: "Please deactivate WooCommerce Payments."
What do you think? I see four options here:

- Leave it as it is
- Make the title more generic, something like "This plugin needs to be deactivated to collect payments". Unfortunately this doesn't bring as much context as the current one.
- Make the case generic, passing only the deactivating plugin `pluginShouldBeDeactivated(plugin: CardPresentPaymentsPlugin)` causing the edge inconsistent case shown above.
- Make the case double generic passing the plugin that should be kept, something like: `pluginShouldBeDeactivated(plugin: CardPresentPaymentsPlugin, shouldBeKept: CardPresentPaymentsPlugin)`
@joshheald 

> Thinking about this more, perhaps we could
> 
> Merge this PR.
> Mirror it on Android.
> Implement support for letting the user choose + remove the screen introduced in this PR.

I agree with that 👍

Reply:
> What if we simply threw "NotImplementException" in this case - it's a case that should never happen and if it happens we want to know about it ASAP and fix it right? Wdyt?

That makes sense! I Implemented something similar in https://github.com/woocommerce/woocommerce-ios/pull/6892/commits/dd0ad80cdf9e33354d1c958a55f34d7e67a77749, checking that the plugin to be deactivated is stripe before showing the view. Otherwise, it goes to the default onboarding view `InPersonPaymentsUnavailable`, that should never happen for the case we have here.
Thanks a lot @malinajirka for the feedback here 🚀